### DISCUSSION TITLE
Column visibility - Is there a way to disable toggleHiddenColumn option for certain columns

### DISCUSSION THREAD

**Discussion by nanandn** [TanStack/table#2817] [Q&A]
Is there a way to disable toggleHiddenColumn option for certain columns. I want to disable column hiding for the "row selection" column and some sticky/frozen columns I have in the table. I have implemented a column chooser drop-down, which contains a "All". The onChange prop provided by getToggleHideAllColumnsProps seems to hide all the columns. How can I control which columns should be hidden?

Thanks
Anand

**Comment by ludvik4** [comment]
[enableHiding](https://tanstack.com/table/latest/docs/guide/column-visibility#disable-hiding-columns)
